birdhouse/config/jupyterhub/jupyterhub_config.py.template

```diff
@@ -25,6 +25,15 @@ c.JupyterHub.db_url = '/persist/jupyterhub.sqlite'
 c.JupyterHub.template_paths = ['/custom_templates']
 
 class CustomDockerSpawner(DockerSpawner):
+    @property
+    def escaped_name(self):
+        """
+        Return the username without escaping. This ensures that mounted directories on the
+        host machine are discovered properly since we expect the username to match the username
+        set by Magpie.
+        """
+        return self.user.name
```

Comment:
Oh I am not sure I want to override `escape_name()`.  This will make `{username}` for old existing username that potentially can have invalid char actually use the raw username with invalid char.

`MAGPIE_USER_NAME_EXTRA_REGEX` take effect on new username only.  Existing ones, if they are invalid, are still invalid.

Reply:
Any user who would have that issue will not be able to spawn a jupyterlab server right now and you would know about it already.

If we don't override `escape_name` then we have to limit the usernames to lowercase ascii and digits. I believe that @fmigneault is still not in favour of that strategy though.

Reply:
The Jupyter usernames were always bounded by Magpie usernames at least. The only mismatch this escape caused was about lower/upper case (which won't even be a thing anymore with an upcoming Magpie PR), and `.-_` characters.

Reply:
> Any user who would have that issue will not be able to spawn a jupyterlab server right now and you would know about it already.

Agreed.  For existing user that never log into Jupyter before, if they generate an error, hopefully they will contact us and we'll create them a new username that is "valid".

Now that I think about it though, isn't overriding `escape_name()` to return `self.user.name` the same as simply using `{raw_username}`?  Why overriding when we can simply use `{raw_username}`?